modules/document_repository/jsx/editForm.js

```diff
@@ -0,0 +1,251 @@
+import PropTypes from 'prop-types';
+/**
+ * Document Edit Form
+ *
+ * Fetches data corresponding to a given file from Loris backend and
+ * displays a form allowing meta information of the media file
+ *
+ * @author Shen Wang
+ * @version 1.0.0
+ *
+ * */
+class DocEditForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      Data: {},
+      docData: {},
+      uploadResult: null,
+      isLoaded: false,
+      loadedData: 0,
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.showAlertMessage = this.showAlertMessage.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  fetchData() {
+    return fetch(this.props.DataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => {
+      let formData = {
+          idDocFile: data.docData.id,
+          pscid: data.docData.pscid,
+          category: data.docData.category,
+          visitLabel: data.docData.visitLabel,
+          comments: data.docData.comments,
+          version: data.docData.version,
+          instrument: data.docData.instrument,
+          category: data.docData.category,
+          forSite: data.docData.forSite,
+        };
+        this.setState({
+          Data: data,
+          docData: data.docData,
+          formData: formData,
+        });
+      })
+      .catch((error) => {
+        this.setState({error: true});
+    });
+  }
+
+  render() {
+    // Data loading error
+    if (this.state.error !== undefined) {
+      return (
+        <div className="alert alert-danger text-center">
+          <strong>
+            {this.state.error}
+          </strong>
+        </div>
+      );
+    }
+
+    // Waiting for data to load
+    if (!this.state.isLoaded) {
+      return (
+        <button className="btn-info has-spinner">
+          Loading
+          <span
+            className="glyphicon glyphicon-refresh glyphicon-refresh-animate">
+          </span>
+        </button>
+      );
+    }
+
+    let alertMessage = '';
+    let alertClass = 'alert text-center hide';
+    let backURL = loris.BaseURL.concat('/document_repository/');
+
+    if (this.state.uploadResult) {
+      if (this.state.uploadResult === 'success') {
+        alertClass = 'alert alert-success text-center';
+        alertMessage = 'Update Successful!';
+      } else if (this.state.uploadResult === 'error') {
+        alertClass = 'alert alert-danger text-center';
+        alertMessage = 'Failed to update the file';
+      }
+    }
+
+    return (
+      <div>
+        <div className={alertClass} role="alert" ref="alert-message">
+          {alertMessage}
+        </div>
+        {
+          this.state.uploadResult === 'success' ?
+          <a className="btn btn-primary" href={backURL}>Back to document repository</a> :
+          null
+        }
```

Comment:
Instead of alert messages, you should be using `swals`. I suggest reworking lines `85`-`97` and `101` to `108`.